### DISCUSSION TITLE
Update/Upload MetaData with CID

### DISCUSSION THREAD

**Discussion by htafolla** [TENK-DAO/tenk#74] [Q&A]
There appears to be a chicken and egg scenario with updating the .json `media` metadata with the correct CID when both the png/json files are stored and uploaded from the same directory to IPFS.

https://github.com/TENK-DAO/tenk#uploading-assets-with-nft-cli

Can you please provide the proper process to have the json files updated with the proper CID. Do both the json and png files need to be stored under the same CID as the README indicates?

In the case that the json files can be stored under a different CID is there a linking process or hash needed to provide full support for NEP-171?

**Comment by willemneal** [comment]
Since they are both found with the same CID, `<CID>/0.png`, `<CID>/0.json`, I've found it unnecessary to provide the CID field in the JSON.  That is, you found it with the CID so you can find the corresponding `png`.  Since consumers of the NFT typically use the base_uri + media or reference fields, the metadata having the the IPFS reference to the png isn't needed.

However, to answer question, the way to do this is to first have a folder with just media, e.g. `<dir>/0.png` , `<dir>/1.png`.  Then use `nft-cli` to pack and upload it.

`nft pack <dir>`, provides the corresponding media CID.  Then update each of the metadata json files with this `CID/0.png`, etc.  

Lastly you combine the files into one folder and provide as the README suggests.

Perhaps this should be one step, but I didn't want to make any assumptions for the shape of the metadata since it's not standard.
